nesta/packages/date_utils/date_utils.py

```diff
@@ -0,0 +1,28 @@
+from dateutil import rrule
+from datetime import datetime as dt
+from datetime import timedelta
+from pandas import Timestamp
+
+def weekchunks(start, until=None, until_days_ago=0, 
+               date_format='%Y-%m-%d'):
+    '''Generate date strings in weekly chunks between two dates.
+    Args:
+        start (str): Sensibly formatted datestring (format to be guessed by pd)
+        until (str): Another datestring. Default=today.
+        until_days_ago (str): if until is not specified, this indicates how many days ago to consider. Default=0.
+                              if until is specified, this indicates an offset of days before the until date.
+    Returns:
+        chunk_pairs (list): List of pairs of string, representing the start and end of weeks.
+    '''
+    until = Timestamp(until).to_pydatetime() if until is not None else dt.now() - timedelta(days=until_days_ago)
+    start = Timestamp(start).to_pydatetime()
+    chunks = [dt.strftime(_date, date_format)
+              for _date in rrule.rrule(rrule.WEEKLY, dtstart=start,
+                                       until=until)]
+    if len(chunks) == 1:  # the less-than-one-week case
+        _until = dt.strftime(until, date_format)
+        chunks.append(_until)
+    chunk_pairs = []
+    for i in range(len(chunks)-1):
+        chunk_pairs.append((chunks[i], chunks[i+1]))
+    return chunk_pairs
```

Comment:
Note: Could use https://toolz.readthedocs.io/en/latest/api.html#toolz.itertoolz.sliding_window (or perhaps I like `toolz` a little too much :sweat_smile: )

Reply:
nah, you like it the right amount. I've made the switch!